client/packages/common/src/ui/components/inputs/Filters/DateFilter.tsx

```diff
@@ -44,10 +42,15 @@ export const DateFilter = ({
         return;
       }
 
-      const date =
-        range === 'to' && displayAs === 'date' // if no time picker, set "TO" field to end of day
-          ? DateUtils.endOfDay(selection)
-          : selection;
+      let date = selection;
+
+      if (!showTimepicker && type === 'dateTime') {
```

Comment:
this one might be helped by a little comment... something like "when user is only selecting the date, ensure our search filter is inclusive of start and end day" ?